org/teams.yaml

```diff
@@ -205,20 +205,6 @@ teams:
           go-control-plane: write
           proxy: write
         teams:
-          WG - Networking Maintainers/Data Plane:
```

Comment:
LGTM to consolidate the list, but this is essentially removing the following people who are at least potentionally active:

@JimmyCYJ @linsun @venilnoronha 

folks tagged, if you would like to remain maintainers of istio/proxy can you join the Policies and Telemetries group?

Reply:
Thanks! I am okay with this change.